data/pools/transactionPool.go

```diff
@@ -695,7 +695,7 @@ func (pool *TransactionPool) recomputeBlockEvaluator(committedTxIds map[transact
 			}
 
 			switch err.(type) {
-			case ledgercore.TransactionInLedgerError:
+			case *ledgercore.TransactionInLedgerError:
```

Comment:
This looks confusing, and can be mistaken somewhere else. 
Why not return the error by value, and make the Txid inside it a pointer, if copying it is a concern?
type TransactionInLedgerError struct {                                                                           
        Txid *transactions.Txid                                                                                   
}

Reply:
I agree. We clearly need to figure this out. However, as you've noted, it could be a sensitive issue where fixing it incorrectly could lead to other issues. That's why I opted in for the simple fix.